libpdbg/sbe_api.c

```diff
@@ -262,14 +281,18 @@ static int sbe_read_state_register(struct pdbg_target *pib, uint32_t *value)
 
 static int sbe_write_state_register(struct pdbg_target *pib, uint32_t value)
 {
-	struct pdbg_target *fsi = pdbg_target_require_parent("fsi", pib);
 	int rc;
 
 	assert(pdbg_target_is_class(pib, "pib"));
 
 	if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
 		return -1;
 
+	//get parent proc target
+	struct pdbg_target *proc = pdbg_target_require_parent("proc", pib);
+	//get fsi traget
+	struct pdbg_target *fsi = sbe_get_fsi_target(proc);
```

Comment:
Declare your variables at the start of the function.

Reply:
ac